### PR TITLE
chore: push files serially to avoid using too many resources

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

**Comment by umutuzgur** [comment]
@imurchie It is not installed yet. It is just left in the file system as stale files and folder. Next time we try to install that app, we first delete the folder in case there are leftovers

File: lib/ios-deploy.js [hunk 1]
```diff
@@ -75,15 +74,14 @@ class IOSDeploy {
           const readStream = fs.createReadStream(itemPath, {autoClose: true});
           const writeStream = await afcService.createWriteStream(pathOnPhone, {autoDestroy: true });
           writeStream.on('finish', writeStream.destroy);
-          promises.push(new B((resolve) => writeStream.on('close', resolve)));
+          const wait = new B((resolve, reject) => {
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: can we have a better name for this promise?

File: lib/ios-deploy.js [hunk 2]
```diff
@@ -75,15 +74,14 @@ class IOSDeploy {
...
+            writeStream.on('close', resolve);
+            writeStream.on('error', reject);
+          });
           readStream.pipe(writeStream);
+          await wait.timeout(`Couldn't push "${itemPath}" within the timeout ${APPLICATION_PUSH_TIMEOUT}ms`, APPLICATION_PUSH_TIMEOUT);
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: we usually use single quotes

File: lib/ios-deploy.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ import _ from 'lodash';
 
 const APPLICATION_INSTALLED_NOTIFICATION = 'com.apple.mobile.application_installed';
 const INSTALLATION_STAGING_DIR = 'PublicStaging';
-const APPLICATION_PUSH_TIMEOUT = 60 * 1000;
+const APPLICATION_PUSH_TIMEOUT = 30 * 1000;
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: this will be item push timeout, not application

**Pull request merged**